### DISCUSSION TITLE
Déterrage de Topics;  Passage de variable entre scènes

### DISCUSSION THREAD

**Discussion by edejean** [witsa/synapps#632] [Q&A] [answered]
### Discussed in https://github.com/witsa/synapps/discussions/519

<div type='discussions-op-text'>

<sup>Originally posted by **JYK-WIT** July  9, 2022</sup>
Bonjour,
J'ai une scène générique qui doit être accessible pas plusieurs autres. Chaque scène doit envoyer au moins une variable à celle-ci.
![image](https://user-images.githubusercontent.com/100723121/178099050-6e160082-60b3-4fc7-bce2-19e9308bcceb.png)
Comment faire ?
Merci
</div>

Bonjour,
Désolé mais le lien vers la réponse complète n'a pas l'air fonctionnel, ou sinon pourriez-vous documenter cette fonction evc des exemples ?
Cdt
Eric

**Comment by Raphwit** [comment]
Bonjour, il est possible de créer ce que l’on appelle un « paramètre de scène ».
Autrement dit on créer sur la scène elle même une additionnelle du type souhaité.
Cette additionnelle pourra ensuite être utilisée par les autres scènes lors de l’utilisation de cette scène dans un acteur écran.

**Comment by edejean** [comment]
Je vais expliquer ce que je veux faire.

Sur ma page principale, je suis sur un système à 3 colonnes, 2 fixes en extérieur, et affichage de mes scènes sur la frame centrale. Sur la droite, un décompte des alarmes par métier, en cliquant, sur un des métiers, je désire faire apparaître dans ma frame centrale ma scène journal avec le filtre équipement correspondant...

Mon bandeau "Alarmes" :

![image](https://github.com/witsa/synapps/assets/105919791/cfeef776-783c-461e-b602-e323420bead7)

Ma scène "journal" avec filtre équipement déjà fonctionnel avec la liste de boutons :

![image](https://github.com/witsa/synapps/assets/105919791/f7105e54-0016-4fd3-ab75-20b95cd853b3)

Je n'arrive pas à comprendre où récupérer ma valeur d'équipement qui par exemple est "E001" pour mon métier "ELEC" et afficher ma scène "Journal en cours" filtrée sur "ELEC"...

**Comment by Raphwit** [comment, marked as answer]
Voici un exemple de la mécanique dont je parle dans mon message précédent :

```
SYNAPPS-STUDIO-SCENE|{"config":{"key":"scene123","name":"Scène 4","additionalDefs":{"prop123":{"type":"text"}}},"leadActor":{"type":"layout/stack","key":"stack1","children":[{"type":"display/screen","key":"screen1","additionals":{"prop1":"ma variable 2.0"},"additionalDefs":{"prop1":{"type":"text"},"prop2":{"type":"text"}}},{"type":"input/button","key":"button1","properties":{"content":"Button"},"events":{"onClick":["context.getActor('screen1').properties.sceneKey = \"scene2\"","context.getActor('screen1').additionals.prop1 = \"hello\""]}}],"bindings":{"properties.backgroundColor":"stage@additionals.prop123"}}}
```

```
SYNAPPS-STUDIO-SCENE|{"config":{"key":"scene2","name":"Scène 2","additionalDefs":{"prop1":{"type":"text"}},"additionals":{"prop1":"ma variable 1.0"}},"leadActor":{"type":"layout/stack","key":"stack1","children":[{"type":"display/text","key":"text1","bindings":{"properties.content":"stage@additionals.prop1"}}]}}
```

Ces deux scènes fonctionnent ensemble.
Dans la première scène, il y a un acteur écran vide (qui n'affiche rien) et qui va âtres renseigné par la pression d'un bouton.
L'évènement 'onClick' de notre bouton va avoir pour effet de renseigner la clé que l'on souhaite à la scène, ainsi que de préciser le paramètre (dans notre cas c'est la 'scene2' et l'additionelle 'prop1')

Dans la seconde scène, on a simplement ajouté à la scène elle même un paramètre qui permet à la scène d'être contrôlé.

Cela me semble correspondra à votre cas de figure.

**Reply by edejean** [reply]
Merci, je me suis inspiré de cet exemple pour mon cas, c'est résolu.
Bonne journée !
